Camunda EE 7.13.1 Support

dbudke opened the pull request:
Support for Camunda EE 7.13.0
Added documentation for H2 support.
Removed undertow dependency since Tomcat is used
Added support for process testing using plain spring container and spring boot
Added support for process test coverage report for non spring boot tests
Added sample-process.bpmn illustrating process scenario testing
Extended readme.md

dbudke committed:
renamed process definition name and process definition id

dbudke committed:
ugraded to Camunda EE 7.13.0
added h2 dependency
added spring-boot-starter-data-jpa dependency due to java.lang.NoClassDefFoundError: javax/activation/DataSource in Java 11
removed undertow dependency

dbudke committed:
Added documentation to enable H2 support
removed quotes in gradle.properties example, since it does not work on Windows.

dbudke committed:
- removed jpa dependency and added spring-boot-start-jdbc
- added camunda testing dependencies

dbudke committed:
renamve job-executer -> job-executor in the project

dbudke committed:
added process testing and process scenario testing

dbudke committed:
add documentation for enabling postgres database support in README.md

dbudke committed:
removed lambda warnings

dbudke committed:
H2 as default database
fixed Java 11 missing dependency for javax/activation/DataSource
Added support for testing with mocks

dbudke committed:
changed visibility

dbudke committed:
Updated README.md

dbudke committed:
Updated README.md and added process testing

dbudke committed:
added camunda spring boot starter test dependency
added light-weight process unit test

dbudke committed:
renamed ProcessUnitTest to ProcessIntegrationTest
Added light-weight process unit tests

hallerch commented on src/test/java/ch/umb/solutions/consulting/camundaspringbootquickstarter/ProcessScenarioTest.java:
```diff
@@ -54,9 +55,7 @@ public void testHappyPath() {
     // when(myProcess.waitsAtReceiveTask(anyString())).thenReturn((messageSubscription) -> {
     //  messageSubscription.receive();
     // });
```
Is the comment needed? If the comment is needed than this code would most likely give a lambda warnings like on the lines 57-59 that you already changed to line 58.

hallerch replied:
//when(myProcess.waitsAtReceiveTask(anyString())).thenReturn(EventSubscriptionDelegate::receive);

dbudke replied:
Updated the test case and added more examples including process test coverage report in target folder

hallerch replied:
When I run the tests for the first time then one test fails (ProcessIntegrationTest.java -> testHapyPath()).
When I run the tests a second time then 4 of 5 tests fails.  
(ProcessIntegrationTest.java -> testHapyPath() and ProcessSenarioTest.java -> testHappyPath(), testAutoApproved(), testManuallyDeclined)

So every second time 4 of 5 tests fails and every second time(+1) 1 of 5 test fails.
Do you have the same behavior?

dbudke replied:
Updated the readme to either user Spring Boot integration tests or the plain process tests.

hallerch reviewed (commented):
Build was successful.
2 of 2 tests passed.
Code looks good for me.
One comment on ProcessScenarioTest.java lines 55 to 57

dbudke committed:
added example test process test1.bpmn (approval process) inclusing scenario testing
enabled process test coverage report for both non-Spring-Boot tests
extended gradle clean task to delete target folder generated by process test coverage plugin

dbudke committed:
added process engine config for non Spring Boot tests

dbudke committed:
target folder not correctly recognized

dbudke committed:
Updated Camunda to 7.13.1
added spring-boot-dev-tools
added service task to sample-process.bpmn
extended README.md

dbudke committed:
configurable Camunda version
extended LoggerDelegate to allow for custom log message
added ProcessConstants.java for process variables

nik-sta committed:
Merge branch 'master' into master
